builder/common/step_run_spot_instance.go

```diff
@@ -383,6 +385,12 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 		Type: aws.String("instant"),
 	}
 
+	if slices.Contains(ec2.SpotAllocationStrategy_Values(), s.SpotAllocationStrategy) {
```

Comment:
The idea makes sense, but I would suggest doing the check in the config's Prepare function, and error if the allocation strategy is not empty and not among the supported values.
Once that check is implemented over here, this can become a simple `if s.SpotAllocationStrategy != ""`.

What worries me with the current implementation is that if someone defines an invalid value for any reason (typo typically comes to mind), Packer will not warn/error of this, and will silently ignore the invalid configuration value, which is something we try to avoid as much as possible.

Reply:
Thanks for pointing out! I've moved that check to prepare step in 807b998. However, should I use the `slice.contains` with the `_Values` method from `ec2` library like the current version, or should change it to a manual string/array check like other checks (for code consistency)?